pkg/operator/controller/garden/garden/components.go

```diff
@@ -651,7 +651,33 @@ func (r *Reconciler) newKubeAPIServer(
 	)
 }
 
-func defaultAPIServerAutoscalingConfig(garden *operatorv1alpha1.Garden) apiserver.AutoscalingConfig {
+func gardenerAPIServerAutoscalingConfig(garden *operatorv1alpha1.Garden) apiserver.AutoscalingConfig {
+	replicas := int32(2)
+	if helper.HighAvailabilityEnabled(garden) {
+		replicas = 3
+	}
+
+	var minAllowed corev1.ResourceList
+	if kubeAPIServer := garden.Spec.VirtualCluster.Kubernetes.KubeAPIServer; kubeAPIServer != nil && kubeAPIServer.Autoscaling != nil {
+		minAllowed = kubeAPIServer.Autoscaling.MinAllowed
+	}
+
+	return apiserver.AutoscalingConfig{
+		APIServerResources: corev1.ResourceRequirements{
+			Requests: kubernetesutils.MaximumResourcesFromResourceList(corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("600m"),
+				corev1.ResourceMemory: resource.MustParse("512Mi"),
+			},
+				minAllowed,
+			),
```

Comment:
Nit: strange formatting (Same below)

```suggestion
			}, minAllowed),
```